Panels/FrameTrackViewer.js

```diff
@@ -1116,51 +1105,187 @@ define([
+
+            const toolBox = Frame.ToolBox.create(
+                Icon.createFA('fa-bars'),
+                Controls.Compound.FixedWidth(Controls.Compound.StandardMargin(
+                    Controls.Compound.GroupVert({separator: 10}, [theFrame.trackControlsGroup, theFrame._popupMenuExtraControlsGroup])
+                ), toolBoxWidth)
+            );
+            theFrame.setToolBox(toolBox);
+
+            theFrame.getLeftPanel = function () {
+                return theLeftPanel;
+            };
+
+            theFrame.getRightPanel = function () {
+                return theRightPanel;
+            };
+
+            theFrame.addCommandSpacer(40);
+
+            theFrame.addCommand({
+                icon: Icon.createFA("fa-search-plus").addDecorator('fa-arrows-h', 'left', 0, 'bottom', -7, 0.6),
+                hint: _TRL("Zoom in")
+            }, function () {
+                const displayWidth = theLeftPanel._width - Module._trackOffsetLeft - Module._trackOffsetRight;
+                theLeftPanel._handleZoom(1.2, displayWidth / 2);
+                theRightPanel._handleZoom(1.2, displayWidth / 2);
+            });
+
+            theFrame.addCommand({
+                icon: Icon.createFA("fa-search-minus").addDecorator('fa-arrows-h', 'left', 0, 'bottom', -7, 0.6),
+                hint: _TRL("Zoom out")
+            }, function () {
+                const displayWidth = theLeftPanel._width - Module._trackOffsetLeft - Module._trackOffsetRight;
+                theLeftPanel._handleZoom(0.8, displayWidth / 2);
+                theRightPanel._handleZoom(0.8, displayWidth / 2);
+            });
+
+            theFrame.addExtraPopupMenuControl = function (ctrl) {
+                theFrame._popupMenuExtraControlsGroup.add(ctrl);
+            };
+
+            theFrame.addLeftTrack = function (track) {
+                addTrack(theFrame, track, theLeftPanel);
+            };
+
+            theFrame.addRightTrack = function (track) {
+                addTrack(theFrame, track, theRightPanel);
+            };
+
             return theFrame;
         };
 
+        function addTrack(iFrame, iTrack, iPanel) {
+            const $El = $('#' + iPanel.getId() + '_content');
+            const isLive = iPanel._isrunning;
+            iPanel.addTrack(iTrack);
+
+            if (isLive) {
+                $El.append(iTrack.render());
+                iTrack.attachEventHandlers();
+            }
+
+            if (iTrack.canHide()) {
+                iTrack.__ctrl_visible = Controls.Check({text: iTrack.getName(), checked: iTrack.isVisible()});
+                iFrame.trackControlsGroup.add(iTrack.__ctrl_visible);
+                iFrame.trackControlsGroup.liveUpdate();
+                iTrack.__ctrl_visible.addNotificationHandler(function () {
+                    iTrack.setVisible(iTrack.__ctrl_visible.getValue());
+                    iPanel.rescale({resizing: false});
+                });
+            }
```

Comment:
No need to change this now, but this is the only part that might need some changes later. Currently each track is added to the filter menu with the same name (so each track name occurs twice if you show the same viewer left and right). I am not sure what the desired behaviour will be:

- single entries that shows/hides that track in both the left and right viewer (that would require both viewers to have the same tracks),
- or append 'left' and 'right' to the track names, to know which one it is. 

The show/hide tracks does work, so for now I think we can leave it like this.

Reply:
OK indeed I see! I overlooked it to be honest. I can already add left or right but indeed maybe better to change when it's clear what is desired.